astronomer/providers/amazon/aws/operators/redshift_cluster.py

```diff
@@ -15,6 +16,80 @@
+    :param final_cluster_snapshot_identifier: name of final cluster snapshot
+    :param cluster_status_fetch_interval_seconds: interval seconds to wait for fetching cluster status
+    """
+
+    def __init__(
+        self,
+        *,
+        skip_final_cluster_snapshot: bool = True,
+        final_cluster_snapshot_identifier: Optional[str] = None,
+        cluster_status_fetch_interval_seconds: int = 10,
+        aws_conn_id: str = "aws_default",
+        poll_interval: float = 5.0,
+        **kwargs: Any,
+    ):
+        self.skip_final_cluster_snapshot = skip_final_cluster_snapshot
+        self.final_cluster_snapshot_identifier = final_cluster_snapshot_identifier
+        self.cluster_status_fetch_interval_seconds = cluster_status_fetch_interval_seconds
+        self.aws_conn_id = aws_conn_id
+        self.poll_interval = poll_interval
+        super().__init__(**kwargs)
+
+    def execute(self, context: "Context") -> None:
+        """
+        Logic that the operator uses to correctly identify which trigger to
+        execute, and defer execution as expected.
+        """
+        redshift_hook = RedshiftHook(aws_conn_id=self.aws_conn_id)
+        cluster_state = redshift_hook.cluster_status(cluster_identifier=self.cluster_identifier)
+        if cluster_state == "available":
+            self.defer(
+                timeout=self.execution_timeout,
+                trigger=RedshiftClusterTrigger(
+                    task_id=self.task_id,
+                    polling_period_seconds=self.poll_interval,
+                    aws_conn_id=self.aws_conn_id,
+                    cluster_identifier=self.cluster_identifier,
+                    operation_type="delete_cluster",
+                    skip_final_cluster_snapshot=self.skip_final_cluster_snapshot,
+                    final_cluster_snapshot_identifier=self.final_cluster_snapshot_identifier,
+                    cluster_status_fetch_interval_seconds=self.cluster_status_fetch_interval_seconds,
+                ),
+                method_name="execute_complete",
+            )
+        else:
+            self.log.warning(
+                "Unable to delete cluster since cluster is currently in status: %s", cluster_state
+            )
+
+    def execute_complete(self, context: Dict[str, Any], event: Any = None) -> None:
+        """
+        Callback for when the trigger fires - returns immediately.
+        Relies on trigger to throw an exception, otherwise it assumes execution was
+        successful.
+        """
+        if event:
+            if "status" in event and event["status"] == "error":
+                msg = "{0}: {1}".format(event["status"], event["message"])
+                raise AirflowException(msg)
+            elif "status" in event and event["status"] == "success":
+                self.log.info("%s completed successfully.", self.task_id)
+                self.log.info("Deleted cluster successfully")
+                return None
+        else:
```

Comment:
if you won't receive any event from the trigger that should be an error case, what do you think?

Reply:
Raising an exception now.